src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -400,4 +400,23 @@ describe('ReactServerRendering', function() {
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
   });
+
+  it('warns with a no-op when an async setState or replaceState is triggered', function() {
+    var Component = React.createClass({
+      componentWillMount: function() {
+        setTimeout(() => {
+          this.setState({});
+          this.replaceState({});
+        });
+      },
+      render: function() {
+        return <div onClick={() => {}} />;
+      },
+    });
+
+    spyOn(console, 'error');
+    ReactServerRendering.renderToString(<Component />);
+    jest.runAllTimers();
+    expect(console.error.calls.count()).toBe(2);
```

Comment:
It is not clear from a test what the warning is. Can you please assert the warning message as well?